src/System.Management.Automation/engine/ExperimentalFeature/ExperimentalFeature.cs

```diff
@@ -119,11 +119,11 @@ static ExperimentalFeature()
         /// <summary>
         /// Process the array of enabled feature names retrieved from configuration.
         /// Ignore invalid feature names and unavailable engine feature names, and
-        /// return an ImmutableHashSet of the valid enabled feature names.
+        /// return an ReadOnlyHashSet of the valid enabled feature names.
         /// </summary>
-        private static ImmutableHashSet<string> ProcessEnabledFeatures(string[] enabledFeatures)
+        private static ReadOnlyHashSet<string> ProcessEnabledFeatures(string[] enabledFeatures)
```

Comment:
>ReadOnlyHashSet [](start = 23, length = 15)

Why not `ReadOnlyDictionary`? You could avoid adding a new type that way.

Reply:
The `HashSet` is exposed to user via `$EnabledExperimentalFeatures`. I think it's weird to have a `ReadOnlyDictionary` object for this variable.

Reply:
I think it's weird to have `System.Management.Automation.ReadOnlyHashSet`.

How about making it a `HashSet` that is just a clone, or cloned on every access?

Reply:
How about changing the type to `internal`? Since the type has public members, changing it to internal doesn't affect users from using `$EnabledExperimentalFeatures` the way they want (access `Count`, call `Contains(string)`).